backend/api/models/category.py

```diff
@@ -23,57 +23,68 @@ class Meta:
     label = UnicodeAttribute()
     last_updated_timestamp = NumberAttribute(default_for_new=get_current_timestamp)
 
-    @classmethod
-    def _get_category_from_model(cls, category):
-        return {
-            "label": category.label,
-            "dimension": category.dimension,
-            "hierarchy": category.hierarchy,
-        }
-
-    @classmethod
-    def get_category(cls, name: str, version: str):
-        """
-        Gets the category data for a particular category and EDAM version.
-        """
-
-        category = []
-        start = time.perf_counter()
-
-        for item in cls.query(
-            slugify(name), cls.version_hash.startswith(f"{version}:")
-        ):
-            category.append(cls._get_category_from_model(item))
-
-        print_perf_duration(start, f"CategoryModel.get_category({name})")
-
-        return category
-
-    @classmethod
-    def get_all_categories(cls, version: str):
-        """
-        Gets all available category mappings from a particular EDAM version.
-        """
-
-        start = time.perf_counter()
-        categories = cls.scan(
-            cls.version == version,
-            attributes_to_get=[
-                "formatted_name",
-                "version",
-                "dimension",
-                "hierarchy",
-                "label",
-            ],
+    def __eq__(self, other):
+        return isinstance(other, CategoryModel) and (
+            self.name == other.name
+            and self.version_hash == other.version_hash
+            and self.version == other.version
+            and self.formatted_name == other.formatted_name
+            and self.dimension == other.dimension
+            and self.hierarchy == other.hierarchy
+            and self.label == other.label
         )
 
-        mapped_categories = defaultdict(list)
 
-        for category in categories:
-            mapped_categories[category.formatted_name].append(
-                cls._get_category_from_model(category)
-            )
+def _get_category_from_model(category):
+    return {
+        "label": category.label,
+        "dimension": category.dimension,
+        "hierarchy": category.hierarchy,
+    }
+
+
+def get_category(name: str, version: str):
+    """
+    Gets the category data for a particular category and EDAM version.
+    """
+
+    category = []
+    start = time.perf_counter()
+
+    for item in CategoryModel.query(
+        slugify(name), CategoryModel.version_hash.startswith(f"{version}:")
+    ):
+        category.append(_get_category_from_model(item))
+
```

Comment:
Nit: We could do this as follows leveraging some pythonic magic.

```suggestion
    results = CategoryModel.query(
        slugify(name), CategoryModel.version_hash.startswith(f"{version}:")
    )
     category = [ _get_category_from_model(item) for item in results ]
```